### PR TITLE
Added ServiceAccount for pre-install-job

### LINKED ISSUE #65: flux2-chart: Pre-Install Hook fails when PSPs are in place
### Describe the bug a clear and concise description of what the bug is.

When you have PSPs or anything equivalent set up in the cluster which denies Pods from running as root, the Installation of the helm chart will fail, because of the failing Pre-install hook (fux-cli seems to be running as root by default). There is currently also no workaround for this (ok, one very dirty of giving the default service account those privileged rights), because postrender scripts are not evaluated for helm hooks because of a bug in helm (https://github.com/helm/helm/issues/7891). 

### What's your helm version?

v3.7.0

### What's your kubectl version?

v1.22.2

### What's the chart version?

flux2-0.8.6

### What happened?

Installation fails

### What you expected to happen?

Installation succeeds 

### How to reproduce it?

1. Setup PSPs wich require runAsNonRoot
2. Install Chart

### Enter the changed values of values.yaml?

NONE

### Enter the command that you execute and failing/misfunctioning.

helm install flux2 fluxcd-community/flux2 -n flux-system

### Anything else we need to know?

There should be some easy solutions for this like 
- running pod as non-root (runAsUser: 10000 should be sufficiant because this pod only executes the one command, which is possible with every uid) like so https://github.com/steveizzle/helm-charts-1/commit/7b5c0c3c44d32b4300914701f0721d2f54c54554
- or the possibility of having a dedicated serviceAccount for the job (which then can be bound to a role with a special psp) like so https://github.com/steveizzle/helm-charts-1/commit/be2a34b54fd95a26dd8039abab8f183c11ee88b0

i could also help implementing it

### PR DESCRIPTION
**Pull request by steveizzle**
Signed-off-by: steveizzle <stefan.riembauer@gmail.com>

#### What this PR does / why we need it:

This PR adds a ServiceAccount for the pre-install-job to have the possibility of binding psp-roles to the job

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #65 

#### Checklist

- [x] [DCO](https://github.com/fluxcd-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] helm-docs are updated
- [x] Helm chart is tested
- [x] Run `make reviewable`

### CODE REVIEW

File: charts/flux2/templates/pre-install-job-serviceaccount.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    app.kubernetes.io/instance: {{ .Release.Namespace | quote }}
+    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    app.kubernetes.io/version: {{ .Chart.AppVersion | quote  }}
+    app.kubernetes.io/part-of: flux
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+  name: "{{ .Release.Name }}-flux-check"
+  annotations: 
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "-10"
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
```
Comment (pierluigilenoci) [inline charts/flux2/templates/pre-install-job-serviceaccount.yaml:14]: Missing LF here

**Review by pierluigilenoci — APPROVED** [review]
I found just a minor issue ;)

**Comment by pierluigilenoci** [comment]
@dwerder could you please take a look?

**Comment by pierluigilenoci** [comment]
So @steveizzle please add the missing "byte" and we will approve the PR 😛

**Comment by steveizzle** [comment]
@pierluigilenoci Added the missing byte :sweat_smile:

**Comment by pierluigilenoci** [comment]
Thx @steveizzle 🚀

**Pull request merged**